### PR TITLE
Changes to get HA pull request approved

### PR DESCRIPTION
**Pull request by JeffLIrion**
These changes would be *extremely* helpful for getting https://github.com/home-assistant/home-assistant/pull/17767 approved.  

Here's a summary of the changes I made.  

* Make `running_apps` a property, not a method
* Add a `sleep` method
* Add `available` property
* Change states from `'play'` and `'pause'` to `'playing'` and `'paused'`
* Make various attributes, properties, and methods public
  * `_screen_on` --> `screen_on`
  * `_awake` --> `awake`
  * `_launcher` --> `launcher`
  * `_settings` --> `settings`
  * `_wake_lock` --> `wake_lock`
  * `_power` --> `power`
* Reorganize the code
* Bump version to 1.0.7

### CODE REVIEW

File: firetv/__init__.py [hunk 1]
```diff
@@ -116,57 +116,215 @@ def connect(self):
                 signer = Signer(self.adbkey)
 
                 # Connect to the device
-                self._adb = adb_commands.AdbCommands().ConnectDevice(serial=self.host, rsa_keys=[signer])
+                self.adb = adb_commands.AdbCommands().ConnectDevice(serial=self.host, rsa_keys=[signer])
             else:
-                self._adb = adb_commands.AdbCommands().ConnectDevice(serial=self.host)
+                self.adb = adb_commands.AdbCommands().ConnectDevice(serial=self.host)
```
Comment (happyleavesaoc) [inline firetv/__init__.py]: Why does the `adb` attribute need to be public?

File: firetv/__init__.py [hunk 2]
```diff
@@ -116,57 +116,215 @@ def connect(self):
...
+
+        return self._send_intent(app, INTENT_LAUNCH)
+
+    def stop_app(self, app):
+        if not self.adb:
+            return None
+
+        return self._send_intent(PACKAGE_LAUNCHER, INTENT_HOME)
+
+    # ======================================================================= #
```
Comment (happyleavesaoc) [inline firetv/__init__.py:143]: The presence of these breaks suggest to me that we should break up this file.

File: firetv/__init__.py [hunk 3]
```diff
@@ -116,57 +116,215 @@ def connect(self):
...
-            return STATE_OFF
-        if self.current_app["package"] == app:
-            return STATE_ON
-        return STATE_OFF
+    @property
+    def current_app(self):
+        current_focus = self._dump("window windows", "mCurrentFocus").replace("\r", "")
+
+        #logging.error("Current: %s", current_focus)
+        #mCurrentFocus = Window{299091cd u0 com.netflix.ninja / com.netflix.ninja.MainActivity}
```
Comment (happyleavesaoc) [inline firetv/__init__.py]: Please remove debug, or convert to `logging.debug`

File: firetv/__init__.py [hunk 4]
```diff
@@ -116,57 +116,215 @@ def connect(self):
                 signer = Signer(self.adbkey)
 
                 # Connect to the device
-                self._adb = adb_commands.AdbCommands().ConnectDevice(serial=self.host, rsa_keys=[signer])
+                self.adb = adb_commands.AdbCommands().ConnectDevice(serial=self.host, rsa_keys=[signer])
```
Comment (happyleavesaoc) [inline firetv/__init__.py]: While we're at it, we should refactor this part along these lines:

```
kwargs = {
    'serial': self.host
}
if self.adbkey:
    kwargs['rsa_keys'] = [Signer[self.key]]
self.adb = adb_commands.AdbCommands().ConnectDevice(**kwargs)
```

File: firetv/__init__.py (hunk 1 above)
Comment (JeffLIrion) [reply firetv/__init__.py]: In the HA component, I check the `adb` attribute in a few places (e.g., https://github.com/JeffLIrion/home-assistant/blob/dev-firetv-adb/homeassistant/components/media_player/firetv.py#L63). The reviewers didn't like that I was accessing a hidden attribute.

Comment (JeffLIrion) [reply firetv/__init__.py]: https://github.com/JeffLIrion/home-assistant/blob/dev-firetv-adb/homeassistant/components/media_player/firetv.py#L101

```python
        except self.exceptions:
            _LOGGER.error('Failed to execute an ADB command; will attempt to '
                          're-establish the ADB connection in the next update')
            returns = None
            self.firetv.adb = None
```

Here, I set the `adb` attribute to `None` in the event of an ADB error, which will trigger a reconnect attempt in the next update.

File: firetv/__init__.py (hunk 3 above)
Comment (JeffLIrion) [reply firetv/__init__.py]: Removed in https://github.com/JeffLIrion/python-firetv/commit/e0e0c24933df74afc7780eda900ce0c75971d1de

File: firetv/__init__.py (hunk 4 above)
Comment (JeffLIrion) [reply firetv/__init__.py]: Fixed in https://github.com/JeffLIrion/python-firetv/commit/1a484423d522fe0780e4d485fa968c9eff1bfc94

File: firetv/__init__.py (hunk 2 above)
Comment (JeffLIrion) [reply firetv/__init__.py:143]: I like it the way it is.  These breaks simply help a user to figure out what methods are available and whether they're low level (like the ADB methods) or high level (like the "key" methods).

File: firetv/__init__.py (hunk 1 above)
Comment (JeffLIrion) [reply firetv/__init__.py]: But if you'd prefer to keep it hidden, then I suggest we define an `available` property as:

```python
    @property
    def available(self):
        """ Check whether the ADB connection is intact. """
        return bool(self._adb)
```

Comment (JeffLIrion) [reply firetv/__init__.py]: I changed it back to being hidden (`_adb`) in https://github.com/happyleavesaoc/python-firetv/pull/65/commits/01b985386a4b2c4551cdd0c6e11f30f10f84cf59.

**Comment by JeffLIrion** [comment]
My Home Assistant PR is ready to go if you release 1.0.7 on pypi!👍

https://github.com/home-assistant/home-assistant/pull/17767#issuecomment-439707242

**Comment by happyleavesaoc** [comment]
`1.0.7` is pushed to pypi

**Comment by JeffLIrion** [comment]
Thank you!!! I should be able to get my HA PR merged now, so I'll stop bugging you with pull requests and release requests!😉

**Pull request merged**